### PR TITLE
new function for insert seed into drop table of e.g. grass

### PR DESCRIPTION
**Pull request by ademant**
The function copies the existing drop table and insert the new seed on Position 1.
The rarity is adjusted, so that items on pos 1is rarer than on pos. 2.

### CODE REVIEW

**Comment by SmallJoker** [comment]
Thanks for your contribution. Please check out https://dev.minetest.net/Lua_code_style_guidelines and adjust your code accordingly

File: mods/farming/api.lua [hunk 1]
```diff
@@ -389,3 +389,34 @@ farming.register_plant = function(name, def)
 	}
 	return r
 end
+
+
+farming.enlarge_drop = function(item_name,def)
```
Comment (SmallJoker) [inline mods/farming/api.lua]: Needs documentation in `game_api.txt`

**Comment by paramat** [comment]
Can you explain more clearly what this does and what the reason for doing this is?
I can't see any problem with how it is currently so can't see any justification for making this more complex.

**Comment by ademant** [comment]
As seen in the farming redo Project: Each time a new seed like barley should appear, the drop table of grass has to be build up. And while inserting barley, the wheat seed (the correct item) has to be considered in building the new drop table. This could lead to Problems, if two different mods want to add grass seed.
Another Feature is that a mod can more easily add a Kind of high Performance wheat and simulate Evolution by inserting the high Performance seed into the drop table of the normal wheat.

**Comment by tenplus1** [comment]
Only the farming mods add seeds to the grass drops and any other mods wishing to amend these would take this into account already when inserting their own.

**Comment by paramat** [comment]
Ok thanks, am neutral for now then.

File: mods/farming/api.lua [hunk 2]
```diff
@@ -389,3 +389,34 @@ farming.register_plant = function(name, def)
...
+  -- if all seeds have same rarity then the first element will drop more often than following elements
+  for i=1,#new_drop do
+    local new_rarity=2^(#new_drop-i)
+    new_drop[i].rarity=new_rarity
+  end
+  -- grab old drop table
+  local old_def=minetest.registered_nodes[item_name].drop
+  old_def.items=new_drop
+  -- override drop table
+  minetest.override_item(item_name,{drop=old_def})
```
Comment (HybridDog) [inline mods/farming/api.lua]: Please add spaces before and after `=`, and after `,`

**Comment by paramat** [comment]
Very low priority, see tenplus1's comment. It is better that any mod adding a new drop to grass does this, as doing so is so rare. So :-1: not justified in my opinion.
Also no support for a year.

**Pull request closed without merging**